### PR TITLE
Move Laplacian spectre from the state

### PR DESCRIPTION
**Pull request by stoprightthere**
- having the `eigenfunctions` and `eigenvalues_laplacian` in the quote unquote "state" makes no sense
- move them to the kernel itself
- leave `state` as an empty dict, mainly for backward-compatibility reasons

### CODE REVIEW

File: geometric_kernels/kernels/feature_maps.py [hunk 1]
```diff
@@ -59,7 +56,7 @@ def _map(X: B.Numeric, params, state, **kwargs) -> B.Numeric:
         )
 
         weights = B.transpose(B.power(spectrum, 0.5))  # [1, M]
-        Phi = state["eigenfunctions"]
+        Phi = kernel.eigenfunctions()
```
Comment (vabor112) [inline geometric_kernels/kernels/feature_maps.py:59]: What about making `eigenfunctions` a property? Calling it as a method seems old-fashioned to me.

File: geometric_kernels/kernels/feature_maps.py [hunk 2]
```diff
@@ -123,7 +120,7 @@ def _map(X: B.Numeric, params, state, key, **kwargs) -> B.Numeric:
                  for jax, and the same random state (generator) for all other backends.
         """
         key, random_phases = space.random(key, num_random_phases)  # [O, D]
-        eigenvalues = state["eigenvalues_laplacian"]
+        eigenvalues = kernel.eigenvalues_laplacian()
```
Comment (vabor112) [inline geometric_kernels/kernels/feature_maps.py:123]: What about making `eigenvalues_laplacian` a property? Calling it as a method seems old-fashioned to me.

File: geometric_kernels/kernels/geometric_kernels.py [hunk 3]
```diff
@@ -62,13 +66,7 @@ def init_params_and_state(self):
...
         """
         params = dict(lengthscale=np.array(1.0), nu=np.array(np.inf))
-
-        eigenvalues_laplacian = self.space.get_eigenvalues(self.num_eigenfunctions)
-        eigenfunctions = self.space.get_eigenfunctions(self.num_eigenfunctions)
-
-        state = dict(
-            eigenvalues_laplacian=eigenvalues_laplacian, eigenfunctions=eigenfunctions
-        )
+        state = dict()
```
Comment (vabor112) [inline geometric_kernels/kernels/geometric_kernels.py:69]: Maybe we should actually use this opportunity to break backwards-compatibility early on (can discuss on the next meeting)

Comment (stoprightthere) [reply geometric_kernels/kernels/geometric_kernels.py:69]: Could you then take over and make sure all the notebook examples are updated accordingly?

Comment (vabor112) [reply geometric_kernels/kernels/geometric_kernels.py:69]: Yes, I am revising the example notebooks anyway

**Comment by stoprightthere** [comment]
Got rid of the state altogether. I didn't update the notebooks/docs though, but I believe a separate PR would be fit for that

**Review by vabor112 — APPROVED** [review]
LGTM

**Pull request merged**